### PR TITLE
bug: efl includes <pthread.h>

### PR DESCRIPTION
**Pull request by caiosignor**
Efl still includes pthreads.h when it should be #include <Einha.h> outside eina/ and #include "eina_threads.h" inside eina/.

With this PR, efl will include <Eina.h> or "eina_threads.h"

### CODE REVIEW

File: src/lib/eina/eina_sched_posix.c [hunk 1]
```diff
@@ -21,7 +21,7 @@
 #include <sys/time.h>
 #include <sys/resource.h>
 #include <errno.h>
-#include <pthread.h>
+
```
Comment (joaoantoniocardoso) [inline src/lib/eina/eina_sched_posix.c:24]: It shouldn't include "eina_thread.h" too?

File: src/examples/elementary/efl_thread_1.c [hunk 2]
```diff
@@ -1,7 +1,7 @@
 //Compile with:
 //gcc -o efl_thread_1 efl_thread_1.c -g `pkg-config --cflags --libs elementary`
 #include <Elementary.h>
-#include <pthread.h>
+#include <eina_thread.h>
```
Comment (joaoantoniocardoso) [inline src/examples/elementary/efl_thread_1.c:4]: I think that if it is not inside eina, it should include <Eina.h> and not directly <eina_thread.h>.

File: src/lib/eina/eina_sched_posix.c (hunk 1 above)
Comment (caiosignor) [reply src/lib/eina/eina_sched_posix.c:24]: it include eina_sched.h, who includes eina_thread.h

File: src/examples/elementary/efl_thread_1.c (hunk 2 above)
Comment (caiosignor) [reply src/examples/elementary/efl_thread_1.c:4]: oh

File: src/lib/eina/eina_sched_posix.c (hunk 1 above)
Comment (joaoantoniocardoso) [reply src/lib/eina/eina_sched_posix.c:24]: Hmm ok. For now it is not clear to me if it is a good practice. Personally I prefer to explicitly include all the needed headers even if it seems redundant..

Comment (JPTIZ) [reply src/lib/eina/eina_sched_posix.c:24]: To be honest, personally I would prefer to include wanted dependencies explicitly, but no big deal for now.

**Pull request closed without merging**